File: week5/deserialization/nodejs/get.js
const http = require('http');

const userData = {
  name: 'Ali',
  age: 25,
  is_admin: true
};

const jsonData = JSON.stringify(userData);

const options = {
  hostname: 'localhost',
  port: 5000,
  path: '/api/user',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': Buffer.byteLength(jsonData)
  }
};

const req = http.request(options, res => {
  let responseData = '';

  res.on('data', chunk => {
    responseData += chunk;
  });

  res.on('end', () => {
    console.log('Server Response:', responseData);
  });
});

req.on('error', error => {
  console.error('Error:', error);
});

req.write(jsonData);
req.end();

File: week5/deserialization/python/get.py
# server.py
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route("/api/user", methods=["POST"])
def receive_user():
    # Deserialize JSON payload
    data = request.get_json()
    
    # Process the data
    name = data.get("name")
    age = data.get("age")
    is_admin = data.get("is_admin")

    print(f"Received user: {name}, Age: {age}, Admin: {is_admin}")

    # Respond back
    return jsonify({"status": "success", "message": f"Hello {name}, your data was received!"})

if __name__ == "__main__":
    app.run(debug=True)
